subsys/bluetooth/mesh/rpl.h

```diff
@@ -27,3 +27,4 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 void bt_mesh_rpl_clear(void);
 void bt_mesh_rpl_update(struct bt_mesh_rpl *rpl,
 			struct bt_mesh_net_rx *rx);
+uint16_t bt_mesh_rpl_list_get(struct bt_mesh_rpl **rpl_list);
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/pull/43441#discussion_r820671571

It is what I did already and this was claimed as a security violation. I guess rpl shouldn't provide direct access to rpl list outside itself.
rpl list size is Kconfig option and is not changed in real time. No necessity to return it as part of API. IMHO

Reply:
I agree that returning the size of the RPL might be unessesary 👍 

When it comes to the direct access to the RPL, Trond wrote this:

> If an application wishes to make their own storage mechanism for RPL, this should be considered a part of the mesh stack, and not part of the application, and it should use an internal API to achieve it. I'm heavily against putting anything like this in include/

I guess we can consider this an internal API? In either case the EMDS is dependent on having access to the entire memory area, so that leaves us with three options as far as I can see. Either the RPL storage module must have access to the RPL through an API (Like implemented), the RPL storage module must own the memory area the RPL is stored in, or the list must be duplicated in both modules.

My initial thought is that the internal API is the best solution, unless I'm missing some other prudent option for solving this. Any thoughts?

Reply:
Could we do it in the opposite manner? RPL should use the list of methods from the storage module to store rpl instances.
Vulnerability is not a question of internal or public API. There is some externally available function that you can invoke to get full access to the security table.  
Something what settings subsystem has. IMHO Settings is not ideal but this doesn't add the vulnerability.

Reply:
You mean declaring the RPL list inside the storage module? And the use some sort of `foreach()`, `put()`, `get()` etc. from rpl.c to access the memory area?

Reply:
How about this:
```
- in rpl_storage.c/.h
    - Move replay_list[] from rpl.c;
    - Move BT_MESH_SETTINGS_SET, rpl_set(), store_rpl(), clear_rpl(), bt_mesh_rpl_find()/_alloc() from rpl.c;
    - Rename:
        - store_rpl() -> bt_mesh_rpl_strg_store(); Declare in header file;
        - clear_rpl() -> bt_mesh_rpl_strg_clear(); Declare in header file;
    - Add struct bt_mesh_rpl *bt_mesh_rpl_strg_iter(struct bt_mesh_rpl *rpl); -> Returns first entry if rpl is NULL, otherwise, returns next entry; Declare in header file;
- in rpl.c/.h:
    - Leave only bt_mesh_rpl_check()/_reset()/_clear()/_check();
    - Refactor to access replay_list through bt_mesh_rpl_strg_iter();
    - Stores/clears entries in storage as is;

Thus:
- No need to maintain most of rpl.h API;
- No need to give access to replay_list[];
- Can completely replace rpl_storage.c in downstream and re-implementing bt_mesh_rpl_iter();
```

Reply:
Well, as far as I can see I think that could work Pavel 👍

Reply:
I've done the alterations to the best of my understanding. Have another look and see if it corresponds to what you imagined.